### PR TITLE
mkosi: Drop build script workarounds

### PR DESCRIPTION
**Pull request by daandemeyer**

### CODE REVIEW

**Review by medhefgo — APPROVED** [review]
+1 −50 lines changed. Feeling productive today?

**Comment by DaanDeMeyer** [comment]
Added some more stuff so dropping the green label

File: mkosi.postinst [hunk 1]
```diff
@@ -2,19 +2,68 @@
...
         find / -name '*.service' -type f -exec sed -i 's/^\(MemoryDeny\|SystemCall\)/# \1/' {} +
 
         # `systemd-hwdb update` takes > 50s when built with sanitizers so let's not run it by default.
         systemctl mask systemd-hwdb-update.service
     fi
 
+    # Make sure services aren't enabled by default on Debian/Ubuntu.
+    mkdir -p /etc/systemd/system-preset
+    echo "disable *" >/etc/systemd/system-preset/99-mkosi.preset
+
```
Comment (medhefgo) [inline mkosi.postinst]: Intentional? You removed this in the previous commit.

Comment (daandemeyer) [reply mkosi.postinst]: oops

**Comment by medhefgo** [comment]
Maybe convert to bash and use an array for meson args while you're at it?

**Comment by DaanDeMeyer** [comment]
> Maybe convert to bash and use an array for meson args while you're at it?

This is already big enough as it is so let's leave that for another PR

**Review by medhefgo — APPROVED** [review]

**Pull request merged**